src/nexmark/mod.rs

```diff
@@ -36,6 +36,72 @@ pub mod generator;
 pub mod model;
 pub mod queries;
 
+/// BatchedReceiver abstracts the Receiver interface for channels of VecDeque's.
+pub struct BatchedReceiver<T> {
```

Comment:
Something I'm not entirely clear on: Does the order of events matter? If generator 1 creates an event, does it need to be in a specific sequence relative to the events produced  by other generators?

Reply:
> Does the order of events matter? 

Yes,

> If generator 1 creates an event, does it need to be in a specific sequence relative to the events produced by other generators?

Not necessarily - the order of the events produced by the `NexmarkSource` is deterministic  ([example tests](https://github.com/vmware/database-stream-processor/blob/38378b25b92dd7b94daa8898f38b290c224d0194/src/nexmark/mod.rs#L298-L352)), even when explicitly configuring to ensure events are "out of order". How the generation of those events are parallelised so that the `NexmarkSource` can collect and emit them in the correct order is up to us, and is the only factor in the ordering of events from a generator thread relative to the other generator threads.

So without batching, the simplest way was to ensure that the generator threads produce interleaved events, ie. the first of three generators produces events [0, 2, 5, ...], while the second [1, 3, 6, ...] etc. as [demo'd in this test](https://github.com/vmware/database-stream-processor/blob/38378b25b92dd7b94daa8898f38b290c224d0194/src/nexmark/generator/mod.rs#L223-L250). Then the `NexmarkSource` collection thread can simply iterate through the receiving queues to get the events in the correct order (see the test link in the previous paragraph for demo).

With this PR the events are batched when sent from the generator threads to the collecting `NexmarkSource` thread, so they wouldn't need to be interleaved necessarily, but it's still a neat, trivial way to ensure we can pull them off in the correct order, and it appears to be fast (since they're now `pop_front`'d from a vec, rather than received from the channel). So I haven't (yet) seen a need to change the ordering of the events in the generator threads themselves.

Does that make sense? Let me know if I've missed something.